### PR TITLE
Aarch64 tbz patch

### PR DESCRIPTION
**Pull request by adinn**
This patch fixes a problem introduced into AArch64 by a previous truffle compiler fix.

Testing:
The patch fixes all the unit tests which were failing because of a broken tbz implementation.

n.b. 12 CountedLoop tests are still failing but that is due to an unrelated error to do with missing frame state for a SafeSignedDiv node

### CODE REVIEW

File: compiler/src/org.graalvm.compiler.asm.aarch64/src/org/graalvm/compiler/asm/aarch64/AArch64Assembler.java [hunk 1]
```diff
@@ -811,17 +813,74 @@ protected void cbz(int size, Register reg, int imm21, int pos) {
...
+
+    /**
+     * Test a single bit and branch if the bit is nonzero.
+     *
+     * @param reg general purpose register. May not be null, zero-register or stackpointer.
+     * @param uimm6 Unsigned 6-bit bit index.
+     * @param imm16 signed 16 bit offset
+     * @param pos Position at which instruction is inserted into buffer. -1 means insert at end.
+     */
+    protected void tbnz(Register reg, int uimm6, int imm16, int pos) {
```
Comment (axel22) [inline compiler/src/org.graalvm.compiler.asm.aarch64/src/org/graalvm/compiler/asm/aarch64/AArch64Assembler.java]: It seems like you could optionally factor out `tbz` and `tbnz` into a common function called e.g. `testBitAndBranch`, and pass only the encoding to it.

I have no strong feelings, however - as you prefer.

**Comment by axel22** [comment]
LGTM, with an optional comment.

File: compiler/src/org.graalvm.compiler.asm.aarch64/src/org/graalvm/compiler/asm/aarch64/AArch64Assembler.java (hunk 1 above)
Comment (adinn) [inline compiler/src/org.graalvm.compiler.asm.aarch64/src/org/graalvm/compiler/asm/aarch64/AArch64Assembler.java]: Yes, I suppose I could but I don't see it as anything to worry much about.

Comment (axel22) [inline compiler/src/org.graalvm.compiler.asm.aarch64/src/org/graalvm/compiler/asm/aarch64/AArch64Assembler.java]: Sounds great, should land soon then!

**Pull request merged**